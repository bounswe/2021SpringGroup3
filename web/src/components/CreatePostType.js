import React from 'react';
import { Row, Col, Form, Input, Typography, Button, Radio, Select, Space } from 'antd';
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router';
import { CreatePostType as CreatePostTypeRequest } from '../utils/helper';
import { MinusCircleOutlined, PlusOutlined } from '@ant-design/icons';

const types = [
  { label: 'text', value: 'text' },
  { label: 'number', value: 'number' },
  { label: 'link', value: 'link' },
  { label: 'date', value: 'date' },
  { label: 'lacation', value: 'location' },
];

const CreatePostType = (props) => {
  const loginState = useSelector((state) => state.login);
  const navigate = useNavigate()
  const dispatch = useDispatch()
  const [form] = Form.useForm();

  const onFinish = async (values) => {

    values.communityId = props.id;

    console.log('Received values of form:', values);
    values.textFieldNames = values.fields.filter(field => field.type == 'text').map(item => { return item.name })
    values.numberFieldNames = values.fields.filter(field => field.type == 'number').map(item => { return item.name })
    values.linkFieldNames = values.fields.filter(field => field.type == 'link').map(item => { return item.name })
    values.locationFieldNames = values.fields.filter(field => field.type == 'date').map(item => { return item.name })
    values.dateFieldNames = values.fields.filter(field => field.type == 'location').map(item => { return item.name })

    delete values.fields;

    await CreatePostTypeRequest(values, loginState.token, dispatch);
    navigate(`/communities/`)
  }


  return (
    <Row>
      <Col offset={6} span={12} style={{marginTop: "20px"}}>
        <Form name="dynamic_form_nest_item" onFinish={onFinish} autoComplete="off">
          <label>Post Type Name</label>
          <Form.Item
            name="name"
            rules={[
              {
                required: true,
                message: "Post type name cannot be empty!"
              }
            ]}
          >
            <Input placeholder="Post type name" size='middle' />
          </Form.Item>
          <Form.List name="fields">
            {(fields, { add, remove }) => (
              <>
                {fields.map(({ key, name, fieldKey, ...restField }) => (
                  <Col span={24} align="middle">
                    <Space key={key}>
                      <Form.Item
                        {...restField}
                        name={[name, 'type']}
                        fieldKey={[fieldKey, 'type']}
                        rules={[{ required: true, message: 'Missing type' }]}
                      >
                        <Select options={types} style={{ width: 120 }}></Select>
                      </Form.Item>
                      <Form.Item
                        {...restField}
                        name={[name, 'name']}
                        fieldKey={[fieldKey, 'name']}
                        rules={[{ required: true, message: 'Missing field name' }]}
                      >
                        <Input placeholder="Field name"/>
                      </Form.Item>
                      <MinusCircleOutlined onClick={() => remove(name)} />
                    </Space>
                  </Col>
                ))}
                <Form.Item>
                  <Col span={24} align="middle">
                    <Button type="dashed" shape="round" onClick={() => add()} icon={<PlusOutlined />}>
                      Add field
                    </Button>
                  </Col>
                </Form.Item>
              </>
            )}
          </Form.List>
          <Form.Item >
            <Col span={24} align="middle">
              <Button type="primary" shape="round" htmlType="submit">
                Create Post Type
              </Button>
            </Col>
          </Form.Item>
        </Form>
      </Col>
    </Row>
  );
};

export default CreatePostType;