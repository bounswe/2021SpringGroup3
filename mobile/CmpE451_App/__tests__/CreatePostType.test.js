import React from 'react';
import CreatePostType from '../src/screen/CreatePostType';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<CreatePostType />).toJSON();
  expect(tree).toMatchSnapshot();
});
