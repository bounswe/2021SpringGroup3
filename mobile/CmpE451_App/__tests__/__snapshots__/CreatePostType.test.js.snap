// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00227b",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "padding": 7,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 22.5,
                "height": 45,
                "width": 45,
              },
              undefined,
              Object {
                "margin": 0,
              },
            ],
          ]
        }
      >
        <View>
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 30,
                },
                Array [
                  Object {
                    "lineHeight": 30,
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            󰳝
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 4,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 22.5,
                  "height": 45,
                  "width": 45,
                },
                undefined,
                Object {
                  "margin": 0,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  },
                  Array [
                    Object {
                      "lineHeight": 30,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰏗
            </Text>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
            }
          }
        >
          Create Post Type
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <RCTScrollView
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "flex": 1,
        "marginBottom": 20,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "lightgray",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": 10,
            "marginRight": 20,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": 8,
              "paddingTop": 20,
            }
          }
        >
          <TextInput
            autoCapitalize="sentences"
            blurOnSubmit={false}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Post Type Name"
            placeholderTextColor="#8b9cb5"
            returnKeyType="done"
            underlineColorAndroid="#f000"
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3849aa",
              "borderColor": "#3849aa",
              "borderRadius": 12,
              "borderWidth": 0,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "paddingVertical": 10,
                  "textAlign": "center",
                }
              }
            >
              Create
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": 8,
            "paddingTop": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#00227b",
              "fontSize": 16,
              "fontWeight": "600",
              "textAlign": "center",
            }
          }
        >
           Add Text Field 
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 22.5,
                  "height": 45,
                  "width": 45,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#3849aa",
                    "fontSize": 30,
                  },
                  Array [
                    Object {
                      "lineHeight": 30,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰐕
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "lightgray",
              "borderBottomWidth": 1,
              "flex": 9,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": 20,
              "marginRight": 20,
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Enter field name"
            value=""
          />
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18.75,
                  "height": 37.5,
                  "width": 37.5,
                },
                undefined,
                Object {
                  "flex": 1,
                  "marginRight": 12,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#6e74dc",
                    "fontSize": 25,
                  },
                  Array [
                    Object {
                      "lineHeight": 25,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰮈
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": 8,
            "paddingTop": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#00227b",
              "fontSize": 16,
              "fontWeight": "600",
              "textAlign": "center",
            }
          }
        >
          Add Number Field
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 22.5,
                  "height": 45,
                  "width": 45,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#3849aa",
                    "fontSize": 30,
                  },
                  Array [
                    Object {
                      "lineHeight": 30,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰐕
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "lightgray",
              "borderBottomWidth": 1,
              "flex": 9,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": 20,
              "marginRight": 20,
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Enter field name"
            value=""
          />
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18.75,
                  "height": 37.5,
                  "width": 37.5,
                },
                undefined,
                Object {
                  "flex": 1,
                  "marginRight": 12,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#6e74dc",
                    "fontSize": 25,
                  },
                  Array [
                    Object {
                      "lineHeight": 25,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰮈
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": 8,
            "paddingTop": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#00227b",
              "fontSize": 16,
              "fontWeight": "600",
              "textAlign": "center",
            }
          }
        >
           Add Date Field 
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 22.5,
                  "height": 45,
                  "width": 45,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#3849aa",
                    "fontSize": 30,
                  },
                  Array [
                    Object {
                      "lineHeight": 30,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰐕
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "lightgray",
              "borderBottomWidth": 1,
              "flex": 9,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": 20,
              "marginRight": 20,
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Enter field name"
            value=""
          />
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18.75,
                  "height": 37.5,
                  "width": 37.5,
                },
                undefined,
                Object {
                  "flex": 1,
                  "marginRight": 12,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#6e74dc",
                    "fontSize": 25,
                  },
                  Array [
                    Object {
                      "lineHeight": 25,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰮈
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": 8,
            "paddingTop": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#00227b",
              "fontSize": 16,
              "fontWeight": "600",
              "textAlign": "center",
            }
          }
        >
          Add Link Field
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 22.5,
                  "height": 45,
                  "width": 45,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#3849aa",
                    "fontSize": 30,
                  },
                  Array [
                    Object {
                      "lineHeight": 30,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰐕
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "lightgray",
              "borderBottomWidth": 1,
              "flex": 9,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": 20,
              "marginRight": 20,
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Enter field name"
            value=""
          />
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18.75,
                  "height": 37.5,
                  "width": 37.5,
                },
                undefined,
                Object {
                  "flex": 1,
                  "marginRight": 12,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#6e74dc",
                    "fontSize": 25,
                  },
                  Array [
                    Object {
                      "lineHeight": 25,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰮈
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": 8,
            "paddingTop": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#00227b",
              "fontSize": 16,
              "fontWeight": "600",
              "textAlign": "center",
            }
          }
        >
          Add Location Field
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 22.5,
                  "height": 45,
                  "width": 45,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#3849aa",
                    "fontSize": 30,
                  },
                  Array [
                    Object {
                      "lineHeight": 30,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰐕
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "lightgray",
              "borderBottomWidth": 1,
              "flex": 9,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": 20,
              "marginRight": 20,
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Enter field name"
            value=""
          />
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18.75,
                  "height": 37.5,
                  "width": 37.5,
                },
                undefined,
                Object {
                  "flex": 1,
                  "marginRight": 12,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#6e74dc",
                    "fontSize": 25,
                  },
                  Array [
                    Object {
                      "lineHeight": 25,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰮈
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
